### PR TITLE
Editor fetches profiles with resource templates from sinopia server

### LINKED ISSUE #310: retrieve resourceTemplate from server by template id (not server id) within BFF code
depends on having templates in server (LD4P/sinopia_server/issues/38), depends on being able to retrieve them at all (#309), and on having elastic search available to do the mapping from template id to server id (LD4P/sinopia_server/issues/37)

### LINKED ISSUE #373: Upload Profile to Sinopia Server
Clicking the Import Template Button on the `/template` tab will upload the profile to the Sinopia Server. 

Respond with error messages if:
- [ ] User doesn't have rights to upload the profile to particular group

### PR DESCRIPTION
**Pull request by jgreben**
Fixes #373, #309 , #436, #456

- sinopiaServer.js now has function to get and return a non-rdf resource from the sinopia server using the client api
-  reducer checks for existence of `property.valueConstraint` (using `lodash`) before setting output
- `<SinopiaResourceTemplates />` (list of server templates) automatically updates when a new template is uploaded; now shows a message when resource is created on server, and uses the correct template id, label, and key in display table.
- Refactors `<ResourceTemplateForm />` for better testability and code coverage
- `<ResourceTemplate />` when mounted gets the RT from the server based on the given ID. It will display a message if the server is offline or not responsive
- `<PropertyTemplateOutline />` fetches nested resource templates from server. (A lot of whitespace changes here, so please ignore those changes). Refactored case/switch to function for better test coverage. Promise for server fetch happens during button onClick event.
- `<InputLang />` refactored for better test coverage
- `<ImportResourceTemplate />` sets and fulfills the callback functions from its child components `<ImportFileZone />` and `<SinopiaResourceTemplates />` and is the main updating container when a resource template is uploaded. It consolidates the upload messages or error message that displays in `<SinopiaResourceTemplates />` and sends an update key to that component to avoid infinite update recursion.
- `<ImportFileZone />` and `<DropZone />` handle the callbacks to `<ImportResourceTemplate />` that sets the default sinopia server group ('ld4p'). There is commented-out code here for use when we need to have a select form for the user to set the group.
- `<Editor />` takes a resourceTemplateId from either state (in the case of a link from `<SinopiaResourceTemplates />`) or props (in the case of the default Monograph:Instance resource template) and passes it down the form tree. The rendering now waits for the mount and state update to happen and displays a virtually imperceptible message in the meantime.
- `<HomePage />` and `Logout` now passing in the name of the localStorage key being updated (thought we would need this in the future, but login work being done by @jmartin-sul may obviate this)
- `Config` sets the default 'ld4p' group which can be an environment variable. Fake AWS tokens also updated so as not to be a real, albeit outdated JWT.
- `package.json` has the sinopia_server client as a dependency, and ignores the root `index.js` file from code coverage
- Adds tests for all of the above, including some extra tests to increase code coverage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,6 +122,26 @@ Run `docker login` and enter the correct credentials to your docker account.
...
+
+4. Copy everything after the `/` (from `#id_token` to the end) and paste in a temporary text file.
+
+5. Also paste it in as the value for `awsCognitoJWTHashForTest()` in `src/Config.js`
+
+6. From the text file, copy just the `id_token` and paste that in as the value for `awsCognitoIdToken()` in `src/Config.js`
+
+7. Go back to your browser and replace `https://sinopia.io` with your localhost domain (e.g. `localhost:8888`) and hit enter. You are now temporarily authenticated in development for the life of the JWT token.
+
+Note: this process is ticketed to be imporived in upcoming work ticket [https://github.com/LD4P/sinopia_server/issues/89](issues/89)
```
[the remark this answers is not in the document]
Comment (jgreben) [reply README.md]: You know....it's a portmanteau of improved and improvised... 😅

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply README.md]: @jgreben poetry! *chef's kiss*

File: src/Config.js [hunk 2]
```diff
@@ -1,4 +1,12 @@
 class Config {
+  static get sinopiaServerBase() {
+    return process.env.SINOPIA_SERVER_BASE || 'http://localhost:8080'
```
Comment (jermnelson) [inline src/Config.js]: See [comment](https://github.com/LD4P/sinopia_editor/issues/458#issuecomment-484235571) about adding **SINOPIA_SERVER_BASE** to Terraform.

File: src/components/editor/ImportResourceTemplate.jsx [hunk 3]
```diff
@@ -1,43 +1,70 @@
...
+
+  createResourcePromise = (content, group) => new Promise(async (resolve) => {
+    resolve(await instance.createResourceWithHttpInfo(group, content, { slug: content.id, contentType: 'application/json' }))
+  }).then(response => {
+    return response
+  }).catch(() => {})
+
+  fulfillCreateResourcePromise = (promise) => {
+    promise.then(result => {
+      const joinedMessages = this.state.message.slice(0)
```
[the remark this answers is not in the document]
Comment (jgreben) [reply src/components/editor/ImportResourceTemplate.jsx]: `slice(0)` actually returns a new copy of the object in it's current state. I tried `Object.assign({}, this.state.whatever)` and it returns an empty object, which is not what I want.

File: __tests__/components/editor/InputLang.test.js [hunk 4]
```diff
@@ -63,4 +63,20 @@ describe('<InputLang />', () => {
   it('sets the formData store with the total number of objects sent to selected', () => {
     expect(mockFormDataFn.mock.calls.length).toBe(2)
   })
+
+  it('creates a has of options that it renders in the form field', () => {
```
[the remark this answers is not in the document]
Comment (jgreben) [reply __tests__/components/editor/InputLang.test.js]: Fixed!

**Pull request merged**